src/main/java/frc/robot/commands/OneBallAuto.java

```diff
@@ -0,0 +1,26 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.SelectCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Constants;
+import frc.robot.FieldInformation;
+import frc.robot.subsystems.DriveTrain;
+import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.Shooter;
+import frc.robot.subsystems.Vision;
+
+public class OneBallAuto extends SequentialCommandGroup implements AutoCommandInterface {
+
+    public OneBallAuto(Shooter shooter, Intake intake, DriveTrain driveTrain, Vision vision) {
+        addCommands(
+            new TurnTowardsHub(driveTrain, vision).withTimeout(Constants.TIMEOUT_SECS),
+            new SelectCommand(() -> new ShooterCommand(shooter, intake, vision, Shooter.calculateShooterSpeeds(vision.getDistance())))
```

Comment:
Why do you have a SelectCommand here (and all the others)? You are only running a fixed command.

Reply:
I don't like that the speeds are computed outside the command. This will make it hard to have any kind of backup plan if the vision is not working. I suggest we pass in Vision to the command constructor and then have it handle finding the distance and setting the speeds.

Reply:
Actually, it already has "Vision", so just move finding the speeds inside.